### PR TITLE
Lab #8

### PR DESCRIPTION
**Pull request by WilfredoN**

### CODE REVIEW

**Comment by volnsav** [comment]
Did you run your program under valgrind?
Please fix bugs, add short comments to your functions and put them in a separate lib file.

File: lab_08/main.c [hunk 1]
```diff
@@ -0,0 +1,116 @@
+#include <stdio.h>
+#include <time.h>
+#include <stdlib.h>
+
+#define SIZE 8
```
Comment (volnsav) [inline lab_08/main.c]: Please remove it and fix the code.

File: lab_08/main.c [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+{
+    int *array = malloc(sizeof(int) * SIZE);
+    srand(time(NULL));
+    for (int i = 0; i < SIZE; i++)
+    {
+        array[i] = rand() % 49 - 10;
+    }
+    return array;
+}
+
```
Comment (volnsav) [inline lab_08/main.c]: Please separate the declaration from the implementation for all functions.

File: lab_08/main.c [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    for (int i = 0; i < SIZE; i++)
+    {
+        printf("%d ", array[i]);
+    }
+    printf("\n");
+    printf("Count of even numbers in array: %d", countEven(array));
+    printf("\n");
+    printf("First negative element in array: %d", firstNegative(array));
+    printf("\n");
+    printf("Array rotated to the left: \n");
```
Comment (volnsav) [inline lab_08/main.c]: You can include `\n` before and after the message without using separate commands.

**Review by volnsav — CHANGES_REQUESTED** [review]
Please review and address previous and current comments.

File: lab_08/main.c (hunk 1 above)
Comment (capynq) [reply lab_08/main.c]: What do you mean by removing? Check please the new push, I've checked via valgrind, understand the issue (memory leak) and fixed it.

File: lab_08/lib.h [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+int firstNegative(int *array);
+
+// Rotate the array to the left
+void rotateLeft(int *array);
+
+// Sort the array by absolute values
+void sortByAbsolute(int *array);
+
+// Find the median of the array
+float findMedian(int *array);
```
Comment (volnsav) [inline lab_08/lib.h]: it is the wrong approach, in C/C++ you have to pass an array and its size every time.

File: lab_08/lib.h [hunk 5]
```diff
@@ -0,0 +1,30 @@
+
+#include <stdio.h>
+#include <time.h>
+#include <stdlib.h>
+
+#define SIZE 8
```
Comment (volnsav) [inline lab_08/lib.h]: This defies logic. Using a predefined size for such a task limits the applicability of your functions (or library).

**Review by volnsav — CHANGES_REQUESTED** [review]
Please remove the binary files. Then, review and address the comments.

**Review by vlsavch — APPROVED** [review]
Looks good.

**Pull request merged**